catalog/resource_sql_table.go

```diff
@@ -598,18 +598,32 @@ func ResourceSqlTable() common.Resource {
 			// If the user specified a property but the value of that property has changed, that will appear
 			// as a change in the effective property/option. To cause a diff to be detected, we need to
 			// reset the effective property/option to the requested value.
-			userSpecifiedProperties := d.Get("properties").(map[string]interface{})
-			userSpecifiedOptions := d.Get("options").(map[string]interface{})
-			effectiveProperties := d.Get("effective_properties").(map[string]interface{})
-			diff := make(map[string]interface{})
+			userSpecifiedProperties := d.Get("properties").(map[string]any)
+			userSpecifiedOptions := d.Get("options").(map[string]any)
+			effectiveProperties := d.Get("effective_properties").(map[string]any)
+			diff := make(map[string]any)
 			for k, userSpecifiedValue := range userSpecifiedProperties {
 				if effectiveValue, ok := effectiveProperties[k]; !ok || effectiveValue != userSpecifiedValue {
 					diff[k] = userSpecifiedValue
 				}
 			}
-			for k, userSpecifiedValue := range userSpecifiedOptions {
-				if effectiveValue, ok := effectiveProperties["option."+k]; !ok || effectiveValue != userSpecifiedValue {
-					diff["option."+k] = userSpecifiedValue
+			for userOptName, userSpecifiedValue := range userSpecifiedOptions {
+				effectOptName := optionPrefix + userOptName
+				var found bool
+				var effectiveValue any
+				// If the option is not found, check if the user specified the option without the prefix
+				// i.e. if user specified `multiLine` for JSON, then backend returns `spark.sql.dataSourceOptions.multiLine`
+				if effectiveValue, found = effectiveProperties[effectOptName]; !found && !strings.Contains(userOptName, ".") {
+					for k, v := range effectiveProperties {
+						if strings.HasSuffix(k, "."+userOptName) {
```

Comment:
Just checking the suffix feels a bit brittle to me. At least you're checking for `"." + userOptName`, which should be safer. Is `spark.sql.dataSourceOptions.` a well-known prefix for effective options? Would it work if we did something like:
```go
var prefixes = []string{"option.", "spark.sql.dataSourceOptions."}
...
for _, prefix := range prefixes {
        if v, ok := effectiveProperties[prefix + userOptName]; ok {
                found = true
                effectiveValue = v
                break
        }
}
```
This way, we are explicit about what effective property prefixes should be considered as options.

Reply:
it's not clear, really - we need to ask UC team about it.

Reply:
Got confirmation that it's only for data source options so the current version should handle it.